### PR TITLE
Add `Seek::position`

### PR DESCRIPTION
**Pull request by gtors**
Allows obtaining current cursor position in file.

### CODE REVIEW

**Comment by bluss** [comment]
Position is already available via `Seek`. You could also use `.seek()` to implement this.

**Comment by gtors** [comment]
I know, it's possible to use 'seek(SeekFrom::Current(0))', but it's little bit verbose and not obvious.

**Comment by bluss** [comment]
Thanks. Implementation looks good to me this way. The language team will need to discuss adding new API  cc @sfackler @alexcrichton

**Comment by alexcrichton** [comment]
Thanks for the PR @gtors! If we're adding a new API it'll initially land as `#[unstable]` instead of `#[stable]` to give it some time to bake and for us to make tweaks to it as well. Putting this only on `File` seems like it may be a bit odd (to me) because all it requires is a `Seek` implementation. Along those lines this may want to be a default method on the `Seek` trait (and could also be overridden for performance).

The old [I/O RFC](https://github.com/rust-lang/rfcs/blob/master/text/0517-io-os-reform.md#seeking) explicitly indicates that the old `tell` can be replaced with the implementation here (e.g. it was an explicit choice to leave this out), but it was definitely on the conservative side and we can expand.

**Comment by sfackler** [comment]
I think I would be on board with adding this as a defaulted method on `Seek` as well.

**Comment by gtors** [comment]
@alexcrichton @sfackler @bluss So, unstable `Seek::position()` method conflicts with stable `Cursor::position()` method. What should I do?

**Comment by alexcrichton** [comment]
Oh the default trait method would probably return `io::Result<u64>` instead of unwrapping, and I don't think we have to worry too much about it conflicting as inherent methods should take precedent and UFCS can otherwise disambiguate. It is a good point to consider for stabilization though!

File: src/libstd/io/cursor.rs [hunk 1]
```diff
@@ -148,44 +148,22 @@ impl<T> Cursor<T> {
...
-    /// assert_eq!(buff.position(), 0);
-    ///
-    /// buff.seek(SeekFrom::Current(2)).unwrap();
-    /// assert_eq!(buff.position(), 2);
-    ///
-    /// buff.seek(SeekFrom::Current(-1)).unwrap();
-    /// assert_eq!(buff.position(), 1);
-    /// ```
-    #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn position(&self) -> u64 { self.pos }
```
Comment (bluss) [inline src/libstd/io/cursor.rs]: This method can't be removed — it is stable.

**Comment by gtors** [comment]
I'm stuck. Why checks failed?

**Comment by whitequark** [comment]
@gtors  Why did you close this?

**Comment by gtors** [comment]
@whitequark Because I don't know how to implement this task without removing stable Cursor::position method. If there will be both method (Seek::position and Cursor::position), then it may cause confusion. If method will be called differently (suppose ::tell), then the `Cursor` will have two different methods (::position and ::tell) with same functionality.

**Pull request closed without merging**